### ISSUE TITLE
Script monitors ocassionally failing

### ISSUE THREAD

**Issue by sdm216** [wh1te909/tacticalrmm#331]
Since 0.4.24 (Which actually fixed this same issue happening all the time), agents are still occasionally getting the alert below with script checks failing. Repairing the Tactical Agent through Agent Recovery seems to temporarily fix it. With about 200 agents, it's happening about 3-5 devices a day, at random times throughout the day. It keeps failing until the agent is "recovered".

CompanyName, SiteName DeviceName - Script Check: Script Name Failed - Return code: 85
Stdout:
Stderr: open C:\WINDOWS\TEMP\trmm\198018948.ps1: The system cannot find the path specified.

**Comment by dinger1986** [comment]
Have you created the exclusions in your AV?

**Comment by sdm216** [comment]
Yep. I use Defender, and checked the defender event logs, and it isn't blocking anything. And it all works again after recovering the agent, at least for a while.

**Comment by dinger1986** [comment]
Ok. That's interesting I have afew machines that do that as well but always seems to be when they are under load and then it runs fine next time

**Comment by sdm216** [comment]
It seems like the script is being deleted from the temp folder, and the agent isn't automatically re-downloading it until it's recovered. The most recent one was actually not too long after a reboot.

**Comment by dinger1986** [comment]
The way the scripts work in my understanding is they are downloaded and ran each and every time. 

When you said you checked defender logs did you check in event viewer?

**Comment by sdm216** [comment]
Yeah, under Applications & Service Logs> Microsoft > Windows > Windows Defender > Operational. I monitor this log for the events indicating it found something, but I did check manually as well.

**Comment by sdm216** [comment]
Should also note I have all Attack Surface Reduction Rules enabled, but again, it isn't logging that it blocked anything, which it would if it was.

**Comment by dinger1986** [comment]
Ok that's fine, just making sure, like I said I have noticed it. I'll keep an eye on it. @wh1te909 have you seen this happening on your machines?

**Comment by wh1te909** [comment]
ive had this happen only once on 1 agent since I released the fix for the original but. Seems something is deleting the `trmm` folder in C:\Windows\TEMP.

The agent creates this directory everytime the `tacticalagent` windows service starts up, which is why doing a recovery fixes it cuz part of the recovery is to restart that service.

I've fixed this already here https://github.com/wh1te909/rmmagent/commit/67a8ab822c762a819fd65bc8734ba3fd291c547a which will just create the directory if it doesn't exist right before running a script. It will be in the next agent release
